[*] Start mutation process:
   - targets: graph2.py
   - tests: ts-3-5/test_3-5_1-0_326.py
[*] 7 tests passed:
   - test_3-5_1-0_326 [0.19874 s]
[*] Start mutants generation and execution:
   - [#   1] CDI graph2: 
--------------------------------------------------------------------------------
  22:     adjacent(n1, n2): returns True if there is an edge connecting n1 and n2,
  23:     False if not, raises an error if either of the supplied nodes are not in g.
  24:     """
  25:     
- 26:     def __init__(self, data=None):
- 27:         '''Initialize graph.'''
- 28:         self.graph = {}
- 29:         if data:
- 30:             for i in data:
- 31:                 self.add_node(i)
- 32:     
+ 26:     @classmethod
+ 27:     def __init__(self, data=None):
+ 28:         '''Initialize graph.'''
+ 29:         self.graph = {}
+ 30:         if data:
+ 31:             for i in data:
+ 32:                 self.add_node(i)
  33:     def nodes(self):
  34:         '''Return a list of all nodes in the graph.'''
  35:         return list(self.graph.keys())
  36:     
--------------------------------------------------------------------------------
[0.18664 s] survived
   - [#   2] CDI graph2: 
--------------------------------------------------------------------------------
  29:         if data:
  30:             for i in data:
  31:                 self.add_node(i)
  32:     
- 33:     def nodes(self):
- 34:         '''Return a list of all nodes in the graph.'''
- 35:         return list(self.graph.keys())
- 36:     
+ 33:     @classmethod
+ 34:     def nodes(self):
+ 35:         '''Return a list of all nodes in the graph.'''
+ 36:         return list(self.graph.keys())
  37:     def edges(self):
  38:         '''Return a list of all edges in the graph.'''
  39:         return [edge for edges in self.graph.values() for edge in edges]
  40:     
--------------------------------------------------------------------------------
[0.20687 s] survived
   - [#   3] CDI graph2: 
--------------------------------------------------------------------------------
  33:     def nodes(self):
  34:         '''Return a list of all nodes in the graph.'''
  35:         return list(self.graph.keys())
  36:     
- 37:     def edges(self):
- 38:         '''Return a list of all edges in the graph.'''
- 39:         return [edge for edges in self.graph.values() for edge in edges]
- 40:     
+ 37:     @classmethod
+ 38:     def edges(self):
+ 39:         '''Return a list of all edges in the graph.'''
+ 40:         return [edge for edges in self.graph.values() for edge in edges]
  41:     def add_node(self, n):
  42:         '''Add a new node to the graph.'''
  43:         self.graph.setdefault(n, set())
  44:     
--------------------------------------------------------------------------------
[0.18799 s] survived
   - [#   4] CDI graph2: 
--------------------------------------------------------------------------------
  37:     def edges(self):
  38:         '''Return a list of all edges in the graph.'''
  39:         return [edge for edges in self.graph.values() for edge in edges]
  40:     
- 41:     def add_node(self, n):
- 42:         '''Add a new node to the graph.'''
- 43:         self.graph.setdefault(n, set())
- 44:     
+ 41:     @classmethod
+ 42:     def add_node(self, n):
+ 43:         '''Add a new node to the graph.'''
+ 44:         self.graph.setdefault(n, set())
  45:     def add_edge(self, n1, n2):
  46:         '''Add new edge to the graph.'''
  47:         self.graph.setdefault(n1, set([n2]))
  48:         self.graph.setdefault(n2, set())
--------------------------------------------------------------------------------
[0.21173 s] killed by ts-3-5/test_3-5_1-0_326.py::test_add_node

def test_add_node():
        g = Graph()
>       g.add_node('A')

ts-3-5/test_3-5_1-0_326.py:13: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <class 'graph2.Graph'>, n = 'A'

>   ???
E   AttributeError: type object 'Graph' has no attribute 'graph'

graph2:44: AttributeError
   - [#   5] CDI graph2: 
--------------------------------------------------------------------------------
  41:     def add_node(self, n):
  42:         '''Add a new node to the graph.'''
  43:         self.graph.setdefault(n, set())
  44:     
- 45:     def add_edge(self, n1, n2):
- 46:         '''Add new edge to the graph.'''
- 47:         self.graph.setdefault(n1, set([n2]))
- 48:         self.graph.setdefault(n2, set())
- 49:         self.graph[n1].add(n2)
- 50:     
+ 45:     @classmethod
+ 46:     def add_edge(self, n1, n2):
+ 47:         '''Add new edge to the graph.'''
+ 48:         self.graph.setdefault(n1, set([n2]))
+ 49:         self.graph.setdefault(n2, set())
+ 50:         self.graph[n1].add(n2)
  51:     def del_node(self, n):
  52:         """Delete the node 'n' from the graph."""
  53:         del self.graph[n]
  54:         for k in self.graph:
--------------------------------------------------------------------------------
[0.21901 s] killed by ts-3-5/test_3-5_1-0_326.py::test_add_edge

def test_add_edge():
        g = Graph()
        g.add_node('B')
        g.add_node('C')
>       g.add_edge('B', 'C')

ts-3-5/test_3-5_1-0_326.py:20: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <class 'graph2.Graph'>, n1 = 'B', n2 = 'C'

>   ???
E   AttributeError: type object 'Graph' has no attribute 'graph'

graph2:48: AttributeError
   - [#   6] CDI graph2: 
--------------------------------------------------------------------------------
  47:         self.graph.setdefault(n1, set([n2]))
  48:         self.graph.setdefault(n2, set())
  49:         self.graph[n1].add(n2)
  50:     
- 51:     def del_node(self, n):
- 52:         """Delete the node 'n' from the graph."""
- 53:         del self.graph[n]
- 54:         for k in self.graph:
- 55:             self.graph[k].discard(n)
- 56:     
+ 51:     @classmethod
+ 52:     def del_node(self, n):
+ 53:         """Delete the node 'n' from the graph."""
+ 54:         del self.graph[n]
+ 55:         for k in self.graph:
+ 56:             self.graph[k].discard(n)
  57:     def del_edge(self, n1, n2):
  58:         '''Delete the edge connecting n1 and n2.'''
  59:         self.graph[n1].remove(n2)
  60:     
--------------------------------------------------------------------------------
[0.19629 s] killed by ts-3-5/test_3-5_1-0_326.py::test_del_node

def test_del_node():
        g = Graph()
        g.add_node('D')
>       g.del_node('D')

ts-3-5/test_3-5_1-0_326.py:26: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <class 'graph2.Graph'>, n = 'D'

>   ???
E   AttributeError: type object 'Graph' has no attribute 'graph'

graph2:54: AttributeError
   - [#   7] CDI graph2: 
--------------------------------------------------------------------------------
  53:         del self.graph[n]
  54:         for k in self.graph:
  55:             self.graph[k].discard(n)
  56:     
- 57:     def del_edge(self, n1, n2):
- 58:         '''Delete the edge connecting n1 and n2.'''
- 59:         self.graph[n1].remove(n2)
- 60:     
+ 57:     @classmethod
+ 58:     def del_edge(self, n1, n2):
+ 59:         '''Delete the edge connecting n1 and n2.'''
+ 60:         self.graph[n1].remove(n2)
  61:     def has_node(self, n):
  62:         """Return boolean if 'n' is in the graph."""
  63:         return n in self.graph
  64:     
--------------------------------------------------------------------------------
[0.21432 s] killed by ts-3-5/test_3-5_1-0_326.py::test_del_edge

def test_del_edge():
        g = Graph()
        g.add_node('E')
        g.add_node('F')
        g.add_edge('E', 'F')
>       g.del_edge('E', 'F')

ts-3-5/test_3-5_1-0_326.py:34: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <class 'graph2.Graph'>, n1 = 'E', n2 = 'F'

>   ???
E   AttributeError: type object 'Graph' has no attribute 'graph'

graph2:60: AttributeError
   - [#   8] CDI graph2: 
--------------------------------------------------------------------------------
  57:     def del_edge(self, n1, n2):
  58:         '''Delete the edge connecting n1 and n2.'''
  59:         self.graph[n1].remove(n2)
  60:     
- 61:     def has_node(self, n):
- 62:         """Return boolean if 'n' is in the graph."""
- 63:         return n in self.graph
- 64:     
+ 61:     @classmethod
+ 62:     def has_node(self, n):
+ 63:         """Return boolean if 'n' is in the graph."""
+ 64:         return n in self.graph
  65:     def neighbors(self, n):
  66:         '''Return the list of all nodes connected to n by edges.'''
  67:         return self.graph[n]
  68:     
--------------------------------------------------------------------------------
[0.21855 s] killed by ts-3-5/test_3-5_1-0_326.py::test_add_node

def test_add_node():
        g = Graph()
        g.add_node('A')
>       assert g.has_node('A')

ts-3-5/test_3-5_1-0_326.py:14: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <class 'graph2.Graph'>, n = 'A'

>   ???
E   AttributeError: type object 'Graph' has no attribute 'graph'

graph2:64: AttributeError
   - [#   9] CDI graph2: 
--------------------------------------------------------------------------------
  61:     def has_node(self, n):
  62:         """Return boolean if 'n' is in the graph."""
  63:         return n in self.graph
  64:     
- 65:     def neighbors(self, n):
- 66:         '''Return the list of all nodes connected to n by edges.'''
- 67:         return self.graph[n]
- 68:     
+ 65:     @classmethod
+ 66:     def neighbors(self, n):
+ 67:         '''Return the list of all nodes connected to n by edges.'''
+ 68:         return self.graph[n]
  69:     def adjacent(self, n1, n2):
  70:         '''Return boolean if there is an edge connecting n1 and n2.'''
  71:         return n2 in self.neighbors(n1)
--------------------------------------------------------------------------------
[0.21604 s] killed by ts-3-5/test_3-5_1-0_326.py::test_add_edge

def test_add_edge():
        g = Graph()
        g.add_node('B')
        g.add_node('C')
        g.add_edge('B', 'C')
>       assert g.adjacent('B', 'C')

ts-3-5/test_3-5_1-0_326.py:21: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
graph2:71: in adjacent
    ???
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <class 'graph2.Graph'>, n = 'B'

>   ???
E   AttributeError: type object 'Graph' has no attribute 'graph'

graph2:68: AttributeError
   - [#  10] CDI graph2: 
--------------------------------------------------------------------------------
  65:     def neighbors(self, n):
  66:         '''Return the list of all nodes connected to n by edges.'''
  67:         return self.graph[n]
  68:     
- 69:     def adjacent(self, n1, n2):
- 70:         '''Return boolean if there is an edge connecting n1 and n2.'''
- 71:         return n2 in self.neighbors(n1)
+ 69:     @classmethod
+ 70:     def adjacent(self, n1, n2):
+ 71:         '''Return boolean if there is an edge connecting n1 and n2.'''
+ 72:         return n2 in self.neighbors(n1)
--------------------------------------------------------------------------------
[0.21139 s] incompetent

TypeError: graph2:72: TypeError: Graph.neighbors() missing 1 required positional argument: 'n'

   - [#  11] COI graph2: 
--------------------------------------------------------------------------------
  25:     
  26:     def __init__(self, data=None):
  27:         '''Initialize graph.'''
  28:         self.graph = {}
- 29:         if data:
+ 29:         if not data:
  30:             for i in data:
  31:                 self.add_node(i)
  32:     
  33:     def nodes(self):
--------------------------------------------------------------------------------
[0.21565 s] incompetent

TypeError: graph2:30: TypeError: 'NoneType' object is not iterable

   - [#  12] COI graph2: 
--------------------------------------------------------------------------------
  59:         self.graph[n1].remove(n2)
  60:     
  61:     def has_node(self, n):
  62:         """Return boolean if 'n' is in the graph."""
- 63:         return n in self.graph
+ 63:         return n not in self.graph
  64:     
  65:     def neighbors(self, n):
  66:         '''Return the list of all nodes connected to n by edges.'''
  67:         return self.graph[n]
--------------------------------------------------------------------------------
[0.19820 s] killed by ts-3-5/test_3-5_1-0_326.py::test_add_node

def test_add_node():
        g = Graph()
        g.add_node('A')
>       assert g.has_node('A')
E       AssertionError

ts-3-5/test_3-5_1-0_326.py:14: AssertionError
   - [#  13] COI graph2: 
--------------------------------------------------------------------------------
  67:         return self.graph[n]
  68:     
  69:     def adjacent(self, n1, n2):
  70:         '''Return boolean if there is an edge connecting n1 and n2.'''
- 71:         return n2 in self.neighbors(n1)
+ 71:         return n2 not in self.neighbors(n1)
--------------------------------------------------------------------------------
[0.21586 s] killed by ts-3-5/test_3-5_1-0_326.py::test_add_edge

def test_add_edge():
        g = Graph()
        g.add_node('B')
        g.add_node('C')
        g.add_edge('B', 'C')
>       assert g.adjacent('B', 'C')
E       AssertionError

ts-3-5/test_3-5_1-0_326.py:21: AssertionError
   - [#  14] OIL graph2: 
--------------------------------------------------------------------------------
  28:         self.graph = {}
  29:         if data:
  30:             for i in data:
  31:                 self.add_node(i)
- 32:     
+ 32:                 break
  33:     def nodes(self):
  34:         '''Return a list of all nodes in the graph.'''
  35:         return list(self.graph.keys())
  36:     
--------------------------------------------------------------------------------
[0.19081 s] survived
   - [#  15] OIL graph2: 
--------------------------------------------------------------------------------
  52:         """Delete the node 'n' from the graph."""
  53:         del self.graph[n]
  54:         for k in self.graph:
  55:             self.graph[k].discard(n)
- 56:     
+ 56:             break
  57:     def del_edge(self, n1, n2):
  58:         '''Delete the edge connecting n1 and n2.'''
  59:         self.graph[n1].remove(n2)
  60:     
--------------------------------------------------------------------------------
[0.18833 s] survived
   - [#  16] RIL graph2: 
--------------------------------------------------------------------------------
  26:     def __init__(self, data=None):
  27:         '''Initialize graph.'''
  28:         self.graph = {}
  29:         if data:
- 30:             for i in data:
+ 30:             for i in reversed(data):
  31:                 self.add_node(i)
  32:     
  33:     def nodes(self):
  34:         '''Return a list of all nodes in the graph.'''
--------------------------------------------------------------------------------
[0.19060 s] survived
   - [#  17] RIL graph2: 
--------------------------------------------------------------------------------
  50:     
  51:     def del_node(self, n):
  52:         """Delete the node 'n' from the graph."""
  53:         del self.graph[n]
- 54:         for k in self.graph:
+ 54:         for k in reversed(self.graph):
  55:             self.graph[k].discard(n)
  56:     
  57:     def del_edge(self, n1, n2):
  58:         '''Delete the edge connecting n1 and n2.'''
--------------------------------------------------------------------------------
[0.19324 s] survived
   - [#  18] SDI graph2: 
--------------------------------------------------------------------------------
  22:     adjacent(n1, n2): returns True if there is an edge connecting n1 and n2,
  23:     False if not, raises an error if either of the supplied nodes are not in g.
  24:     """
  25:     
- 26:     def __init__(self, data=None):
- 27:         '''Initialize graph.'''
- 28:         self.graph = {}
- 29:         if data:
- 30:             for i in data:
- 31:                 self.add_node(i)
- 32:     
+ 26:     @staticmethod
+ 27:     def __init__(self, data=None):
+ 28:         '''Initialize graph.'''
+ 29:         self.graph = {}
+ 30:         if data:
+ 31:             for i in data:
+ 32:                 self.add_node(i)
  33:     def nodes(self):
  34:         '''Return a list of all nodes in the graph.'''
  35:         return list(self.graph.keys())
  36:     
--------------------------------------------------------------------------------
[0.20360 s] incompetent

TypeError: /home/lucca/teste/python_experiments2/graph2/ts-3-5/test_3-5_1-0_326.py:12: TypeError: Graph.__init__() missing 1 required positional argument: 'self'

   - [#  19] SDI graph2: 
--------------------------------------------------------------------------------
  29:         if data:
  30:             for i in data:
  31:                 self.add_node(i)
  32:     
- 33:     def nodes(self):
- 34:         '''Return a list of all nodes in the graph.'''
- 35:         return list(self.graph.keys())
- 36:     
+ 33:     @staticmethod
+ 34:     def nodes(self):
+ 35:         '''Return a list of all nodes in the graph.'''
+ 36:         return list(self.graph.keys())
  37:     def edges(self):
  38:         '''Return a list of all edges in the graph.'''
  39:         return [edge for edges in self.graph.values() for edge in edges]
  40:     
--------------------------------------------------------------------------------
[0.19085 s] survived
   - [#  20] SDI graph2: 
--------------------------------------------------------------------------------
  33:     def nodes(self):
  34:         '''Return a list of all nodes in the graph.'''
  35:         return list(self.graph.keys())
  36:     
- 37:     def edges(self):
- 38:         '''Return a list of all edges in the graph.'''
- 39:         return [edge for edges in self.graph.values() for edge in edges]
- 40:     
+ 37:     @staticmethod
+ 38:     def edges(self):
+ 39:         '''Return a list of all edges in the graph.'''
+ 40:         return [edge for edges in self.graph.values() for edge in edges]
  41:     def add_node(self, n):
  42:         '''Add a new node to the graph.'''
  43:         self.graph.setdefault(n, set())
  44:     
--------------------------------------------------------------------------------
[0.19337 s] survived
   - [#  21] SDI graph2: 
--------------------------------------------------------------------------------
  37:     def edges(self):
  38:         '''Return a list of all edges in the graph.'''
  39:         return [edge for edges in self.graph.values() for edge in edges]
  40:     
- 41:     def add_node(self, n):
- 42:         '''Add a new node to the graph.'''
- 43:         self.graph.setdefault(n, set())
- 44:     
+ 41:     @staticmethod
+ 42:     def add_node(self, n):
+ 43:         '''Add a new node to the graph.'''
+ 44:         self.graph.setdefault(n, set())
  45:     def add_edge(self, n1, n2):
  46:         '''Add new edge to the graph.'''
  47:         self.graph.setdefault(n1, set([n2]))
  48:         self.graph.setdefault(n2, set())
--------------------------------------------------------------------------------
[0.20908 s] incompetent

TypeError: /home/lucca/teste/python_experiments2/graph2/ts-3-5/test_3-5_1-0_326.py:13: TypeError: Graph.add_node() missing 1 required positional argument: 'n'

   - [#  22] SDI graph2: 
--------------------------------------------------------------------------------
  41:     def add_node(self, n):
  42:         '''Add a new node to the graph.'''
  43:         self.graph.setdefault(n, set())
  44:     
- 45:     def add_edge(self, n1, n2):
- 46:         '''Add new edge to the graph.'''
- 47:         self.graph.setdefault(n1, set([n2]))
- 48:         self.graph.setdefault(n2, set())
- 49:         self.graph[n1].add(n2)
- 50:     
+ 45:     @staticmethod
+ 46:     def add_edge(self, n1, n2):
+ 47:         '''Add new edge to the graph.'''
+ 48:         self.graph.setdefault(n1, set([n2]))
+ 49:         self.graph.setdefault(n2, set())
+ 50:         self.graph[n1].add(n2)
  51:     def del_node(self, n):
  52:         """Delete the node 'n' from the graph."""
  53:         del self.graph[n]
  54:         for k in self.graph:
--------------------------------------------------------------------------------
[0.22119 s] incompetent

TypeError: /home/lucca/teste/python_experiments2/graph2/ts-3-5/test_3-5_1-0_326.py:20: TypeError: Graph.add_edge() missing 1 required positional argument: 'n2'

   - [#  23] SDI graph2: 
--------------------------------------------------------------------------------
  47:         self.graph.setdefault(n1, set([n2]))
  48:         self.graph.setdefault(n2, set())
  49:         self.graph[n1].add(n2)
  50:     
- 51:     def del_node(self, n):
- 52:         """Delete the node 'n' from the graph."""
- 53:         del self.graph[n]
- 54:         for k in self.graph:
- 55:             self.graph[k].discard(n)
- 56:     
+ 51:     @staticmethod
+ 52:     def del_node(self, n):
+ 53:         """Delete the node 'n' from the graph."""
+ 54:         del self.graph[n]
+ 55:         for k in self.graph:
+ 56:             self.graph[k].discard(n)
  57:     def del_edge(self, n1, n2):
  58:         '''Delete the edge connecting n1 and n2.'''
  59:         self.graph[n1].remove(n2)
  60:     
--------------------------------------------------------------------------------
[0.21786 s] incompetent

TypeError: /home/lucca/teste/python_experiments2/graph2/ts-3-5/test_3-5_1-0_326.py:26: TypeError: Graph.del_node() missing 1 required positional argument: 'n'

   - [#  24] SDI graph2: 
--------------------------------------------------------------------------------
  53:         del self.graph[n]
  54:         for k in self.graph:
  55:             self.graph[k].discard(n)
  56:     
- 57:     def del_edge(self, n1, n2):
- 58:         '''Delete the edge connecting n1 and n2.'''
- 59:         self.graph[n1].remove(n2)
- 60:     
+ 57:     @staticmethod
+ 58:     def del_edge(self, n1, n2):
+ 59:         '''Delete the edge connecting n1 and n2.'''
+ 60:         self.graph[n1].remove(n2)
  61:     def has_node(self, n):
  62:         """Return boolean if 'n' is in the graph."""
  63:         return n in self.graph
  64:     
--------------------------------------------------------------------------------
[0.20450 s] incompetent

TypeError: /home/lucca/teste/python_experiments2/graph2/ts-3-5/test_3-5_1-0_326.py:34: TypeError: Graph.del_edge() missing 1 required positional argument: 'n2'

   - [#  25] SDI graph2: 
--------------------------------------------------------------------------------
  57:     def del_edge(self, n1, n2):
  58:         '''Delete the edge connecting n1 and n2.'''
  59:         self.graph[n1].remove(n2)
  60:     
- 61:     def has_node(self, n):
- 62:         """Return boolean if 'n' is in the graph."""
- 63:         return n in self.graph
- 64:     
+ 61:     @staticmethod
+ 62:     def has_node(self, n):
+ 63:         """Return boolean if 'n' is in the graph."""
+ 64:         return n in self.graph
  65:     def neighbors(self, n):
  66:         '''Return the list of all nodes connected to n by edges.'''
  67:         return self.graph[n]
  68:     
--------------------------------------------------------------------------------
[0.21269 s] incompetent

TypeError: /home/lucca/teste/python_experiments2/graph2/ts-3-5/test_3-5_1-0_326.py:14: TypeError: Graph.has_node() missing 1 required positional argument: 'n'

   - [#  26] SDI graph2: 
--------------------------------------------------------------------------------
  61:     def has_node(self, n):
  62:         """Return boolean if 'n' is in the graph."""
  63:         return n in self.graph
  64:     
- 65:     def neighbors(self, n):
- 66:         '''Return the list of all nodes connected to n by edges.'''
- 67:         return self.graph[n]
- 68:     
+ 65:     @staticmethod
+ 66:     def neighbors(self, n):
+ 67:         '''Return the list of all nodes connected to n by edges.'''
+ 68:         return self.graph[n]
  69:     def adjacent(self, n1, n2):
  70:         '''Return boolean if there is an edge connecting n1 and n2.'''
  71:         return n2 in self.neighbors(n1)
--------------------------------------------------------------------------------
[0.21484 s] incompetent

TypeError: graph2:71: TypeError: Graph.neighbors() missing 1 required positional argument: 'n'

   - [#  27] SDI graph2: 
--------------------------------------------------------------------------------
  65:     def neighbors(self, n):
  66:         '''Return the list of all nodes connected to n by edges.'''
  67:         return self.graph[n]
  68:     
- 69:     def adjacent(self, n1, n2):
- 70:         '''Return boolean if there is an edge connecting n1 and n2.'''
- 71:         return n2 in self.neighbors(n1)
+ 69:     @staticmethod
+ 70:     def adjacent(self, n1, n2):
+ 71:         '''Return boolean if there is an edge connecting n1 and n2.'''
+ 72:         return n2 in self.neighbors(n1)
--------------------------------------------------------------------------------
[0.21315 s] incompetent

TypeError: /home/lucca/teste/python_experiments2/graph2/ts-3-5/test_3-5_1-0_326.py:21: TypeError: Graph.adjacent() missing 1 required positional argument: 'n2'

   - [#  28] SDL graph2: 
--------------------------------------------------------------------------------
  24:     """
  25:     
  26:     def __init__(self, data=None):
  27:         '''Initialize graph.'''
- 28:         self.graph = {}
+ 28:         pass
  29:         if data:
  30:             for i in data:
  31:                 self.add_node(i)
  32:     
--------------------------------------------------------------------------------
[0.20706 s] killed by ts-3-5/test_3-5_1-0_326.py::test_add_node

def test_add_node():
        g = Graph()
>       g.add_node('A')

ts-3-5/test_3-5_1-0_326.py:13: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <graph2.Graph object at 0x7f196abb9990>, n = 'A'

>   ???
E   AttributeError: 'Graph' object has no attribute 'graph'

graph2:43: AttributeError
   - [#  29] SDL graph2: 
--------------------------------------------------------------------------------
  27:         '''Initialize graph.'''
  28:         self.graph = {}
  29:         if data:
  30:             for i in data:
- 31:                 self.add_node(i)
+ 31:                 pass
  32:     
  33:     def nodes(self):
  34:         '''Return a list of all nodes in the graph.'''
  35:         return list(self.graph.keys())
--------------------------------------------------------------------------------
[0.20835 s] survived
   - [#  30] SDL graph2: 
--------------------------------------------------------------------------------
  31:                 self.add_node(i)
  32:     
  33:     def nodes(self):
  34:         '''Return a list of all nodes in the graph.'''
- 35:         return list(self.graph.keys())
+ 35:         pass
  36:     
  37:     def edges(self):
  38:         '''Return a list of all edges in the graph.'''
  39:         return [edge for edges in self.graph.values() for edge in edges]
--------------------------------------------------------------------------------
[0.17195 s] survived
   - [#  31] SDL graph2: 
--------------------------------------------------------------------------------
  35:         return list(self.graph.keys())
  36:     
  37:     def edges(self):
  38:         '''Return a list of all edges in the graph.'''
- 39:         return [edge for edges in self.graph.values() for edge in edges]
+ 39:         pass
  40:     
  41:     def add_node(self, n):
  42:         '''Add a new node to the graph.'''
  43:         self.graph.setdefault(n, set())
--------------------------------------------------------------------------------
[0.18793 s] survived
   - [#  32] SDL graph2: 
--------------------------------------------------------------------------------
  39:         return [edge for edges in self.graph.values() for edge in edges]
  40:     
  41:     def add_node(self, n):
  42:         '''Add a new node to the graph.'''
- 43:         self.graph.setdefault(n, set())
+ 43:         pass
  44:     
  45:     def add_edge(self, n1, n2):
  46:         '''Add new edge to the graph.'''
  47:         self.graph.setdefault(n1, set([n2]))
--------------------------------------------------------------------------------
[0.21497 s] killed by ts-3-5/test_3-5_1-0_326.py::test_add_node

def test_add_node():
        g = Graph()
        g.add_node('A')
>       assert g.has_node('A')
E       AssertionError

ts-3-5/test_3-5_1-0_326.py:14: AssertionError
   - [#  33] SDL graph2: 
--------------------------------------------------------------------------------
  43:         self.graph.setdefault(n, set())
  44:     
  45:     def add_edge(self, n1, n2):
  46:         '''Add new edge to the graph.'''
- 47:         self.graph.setdefault(n1, set([n2]))
+ 47:         pass
  48:         self.graph.setdefault(n2, set())
  49:         self.graph[n1].add(n2)
  50:     
  51:     def del_node(self, n):
--------------------------------------------------------------------------------
[0.18670 s] survived
   - [#  34] SDL graph2: 
--------------------------------------------------------------------------------
  44:     
  45:     def add_edge(self, n1, n2):
  46:         '''Add new edge to the graph.'''
  47:         self.graph.setdefault(n1, set([n2]))
- 48:         self.graph.setdefault(n2, set())
+ 48:         pass
  49:         self.graph[n1].add(n2)
  50:     
  51:     def del_node(self, n):
  52:         """Delete the node 'n' from the graph."""
--------------------------------------------------------------------------------
[0.18528 s] survived
   - [#  35] SDL graph2: 
--------------------------------------------------------------------------------
  45:     def add_edge(self, n1, n2):
  46:         '''Add new edge to the graph.'''
  47:         self.graph.setdefault(n1, set([n2]))
  48:         self.graph.setdefault(n2, set())
- 49:         self.graph[n1].add(n2)
+ 49:         pass
  50:     
  51:     def del_node(self, n):
  52:         """Delete the node 'n' from the graph."""
  53:         del self.graph[n]
--------------------------------------------------------------------------------
[0.21509 s] killed by ts-3-5/test_3-5_1-0_326.py::test_add_edge

def test_add_edge():
        g = Graph()
        g.add_node('B')
        g.add_node('C')
        g.add_edge('B', 'C')
>       assert g.adjacent('B', 'C')
E       AssertionError

ts-3-5/test_3-5_1-0_326.py:21: AssertionError
   - [#  36] SDL graph2: 
--------------------------------------------------------------------------------
  51:     def del_node(self, n):
  52:         """Delete the node 'n' from the graph."""
  53:         del self.graph[n]
  54:         for k in self.graph:
- 55:             self.graph[k].discard(n)
+ 55:             pass
  56:     
  57:     def del_edge(self, n1, n2):
  58:         '''Delete the edge connecting n1 and n2.'''
  59:         self.graph[n1].remove(n2)
--------------------------------------------------------------------------------
[0.18271 s] survived
   - [#  37] SDL graph2: 
--------------------------------------------------------------------------------
  55:             self.graph[k].discard(n)
  56:     
  57:     def del_edge(self, n1, n2):
  58:         '''Delete the edge connecting n1 and n2.'''
- 59:         self.graph[n1].remove(n2)
+ 59:         pass
  60:     
  61:     def has_node(self, n):
  62:         """Return boolean if 'n' is in the graph."""
  63:         return n in self.graph
--------------------------------------------------------------------------------
[0.20728 s] killed by ts-3-5/test_3-5_1-0_326.py::test_del_edge

def test_del_edge():
        g = Graph()
        g.add_node('E')
        g.add_node('F')
        g.add_edge('E', 'F')
        g.del_edge('E', 'F')
>       assert not g.adjacent('E', 'F')
E       AssertionError

ts-3-5/test_3-5_1-0_326.py:35: AssertionError
   - [#  38] SDL graph2: 
--------------------------------------------------------------------------------
  59:         self.graph[n1].remove(n2)
  60:     
  61:     def has_node(self, n):
  62:         """Return boolean if 'n' is in the graph."""
- 63:         return n in self.graph
+ 63:         pass
  64:     
  65:     def neighbors(self, n):
  66:         '''Return the list of all nodes connected to n by edges.'''
  67:         return self.graph[n]
--------------------------------------------------------------------------------
[0.21014 s] killed by ts-3-5/test_3-5_1-0_326.py::test_add_node

def test_add_node():
        g = Graph()
        g.add_node('A')
>       assert g.has_node('A')
E       AssertionError

ts-3-5/test_3-5_1-0_326.py:14: AssertionError
   - [#  39] SDL graph2: 
--------------------------------------------------------------------------------
  63:         return n in self.graph
  64:     
  65:     def neighbors(self, n):
  66:         '''Return the list of all nodes connected to n by edges.'''
- 67:         return self.graph[n]
+ 67:         pass
  68:     
  69:     def adjacent(self, n1, n2):
  70:         '''Return boolean if there is an edge connecting n1 and n2.'''
  71:         return n2 in self.neighbors(n1)
--------------------------------------------------------------------------------
[0.21568 s] incompetent

TypeError: graph2:71: TypeError: argument of type 'NoneType' is not iterable

   - [#  40] SDL graph2: 
--------------------------------------------------------------------------------
  67:         return self.graph[n]
  68:     
  69:     def adjacent(self, n1, n2):
  70:         '''Return boolean if there is an edge connecting n1 and n2.'''
- 71:         return n2 in self.neighbors(n1)
+ 71:         pass
--------------------------------------------------------------------------------
[0.21070 s] killed by ts-3-5/test_3-5_1-0_326.py::test_add_edge

def test_add_edge():
        g = Graph()
        g.add_node('B')
        g.add_node('C')
        g.add_edge('B', 'C')
>       assert g.adjacent('B', 'C')
E       AssertionError

ts-3-5/test_3-5_1-0_326.py:21: AssertionError
   - [#  41] SVD graph2: 
--------------------------------------------------------------------------------
  24:     """
  25:     
  26:     def __init__(self, data=None):
  27:         '''Initialize graph.'''
- 28:         self.graph = {}
+ 28:         graph = {}
  29:         if data:
  30:             for i in data:
  31:                 self.add_node(i)
  32:     
--------------------------------------------------------------------------------
[0.00000 s] incompetent

ValueError: expression must have Store context but has Load instead

   - [#  42] SVD graph2: 
--------------------------------------------------------------------------------
  27:         '''Initialize graph.'''
  28:         self.graph = {}
  29:         if data:
  30:             for i in data:
- 31:                 self.add_node(i)
+ 31:                 add_node(i)
  32:     
  33:     def nodes(self):
  34:         '''Return a list of all nodes in the graph.'''
  35:         return list(self.graph.keys())
--------------------------------------------------------------------------------
[0.18847 s] survived
   - [#  43] SVD graph2: 
--------------------------------------------------------------------------------
  31:                 self.add_node(i)
  32:     
  33:     def nodes(self):
  34:         '''Return a list of all nodes in the graph.'''
- 35:         return list(self.graph.keys())
+ 35:         return list(graph.keys())
  36:     
  37:     def edges(self):
  38:         '''Return a list of all edges in the graph.'''
  39:         return [edge for edges in self.graph.values() for edge in edges]
--------------------------------------------------------------------------------
[0.21266 s] survived
   - [#  44] SVD graph2: 
--------------------------------------------------------------------------------
  35:         return list(self.graph.keys())
  36:     
  37:     def edges(self):
  38:         '''Return a list of all edges in the graph.'''
- 39:         return [edge for edges in self.graph.values() for edge in edges]
+ 39:         return [edge for edges in graph.values() for edge in edges]
  40:     
  41:     def add_node(self, n):
  42:         '''Add a new node to the graph.'''
  43:         self.graph.setdefault(n, set())
--------------------------------------------------------------------------------
[0.18364 s] survived
   - [#  45] SVD graph2: 
--------------------------------------------------------------------------------
  39:         return [edge for edges in self.graph.values() for edge in edges]
  40:     
  41:     def add_node(self, n):
  42:         '''Add a new node to the graph.'''
- 43:         self.graph.setdefault(n, set())
+ 43:         graph.setdefault(n, set())
  44:     
  45:     def add_edge(self, n1, n2):
  46:         '''Add new edge to the graph.'''
  47:         self.graph.setdefault(n1, set([n2]))
--------------------------------------------------------------------------------
[0.20921 s] killed by ts-3-5/test_3-5_1-0_326.py::test_add_node

def test_add_node():
        g = Graph()
>       g.add_node('A')

ts-3-5/test_3-5_1-0_326.py:13: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <graph2.Graph object at 0x7f196acc61a0>, n = 'A'

>   ???
E   NameError: name 'graph' is not defined

graph2:43: NameError
   - [#  46] SVD graph2: 
--------------------------------------------------------------------------------
  43:         self.graph.setdefault(n, set())
  44:     
  45:     def add_edge(self, n1, n2):
  46:         '''Add new edge to the graph.'''
- 47:         self.graph.setdefault(n1, set([n2]))
+ 47:         graph.setdefault(n1, set([n2]))
  48:         self.graph.setdefault(n2, set())
  49:         self.graph[n1].add(n2)
  50:     
  51:     def del_node(self, n):
--------------------------------------------------------------------------------
[0.21048 s] killed by ts-3-5/test_3-5_1-0_326.py::test_add_edge

def test_add_edge():
        g = Graph()
        g.add_node('B')
        g.add_node('C')
>       g.add_edge('B', 'C')

ts-3-5/test_3-5_1-0_326.py:20: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <graph2.Graph object at 0x7f196ab6d2d0>, n1 = 'B', n2 = 'C'

>   ???
E   NameError: name 'graph' is not defined

graph2:47: NameError
   - [#  47] SVD graph2: 
--------------------------------------------------------------------------------
  44:     
  45:     def add_edge(self, n1, n2):
  46:         '''Add new edge to the graph.'''
  47:         self.graph.setdefault(n1, set([n2]))
- 48:         self.graph.setdefault(n2, set())
+ 48:         graph.setdefault(n2, set())
  49:         self.graph[n1].add(n2)
  50:     
  51:     def del_node(self, n):
  52:         """Delete the node 'n' from the graph."""
--------------------------------------------------------------------------------
[0.20934 s] killed by ts-3-5/test_3-5_1-0_326.py::test_add_edge

def test_add_edge():
        g = Graph()
        g.add_node('B')
        g.add_node('C')
>       g.add_edge('B', 'C')

ts-3-5/test_3-5_1-0_326.py:20: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <graph2.Graph object at 0x7f196a9735e0>, n1 = 'B', n2 = 'C'

>   ???
E   NameError: name 'graph' is not defined

graph2:48: NameError
   - [#  48] SVD graph2: 
--------------------------------------------------------------------------------
  45:     def add_edge(self, n1, n2):
  46:         '''Add new edge to the graph.'''
  47:         self.graph.setdefault(n1, set([n2]))
  48:         self.graph.setdefault(n2, set())
- 49:         self.graph[n1].add(n2)
+ 49:         graph[n1].add(n2)
  50:     
  51:     def del_node(self, n):
  52:         """Delete the node 'n' from the graph."""
  53:         del self.graph[n]
--------------------------------------------------------------------------------
[0.20887 s] killed by ts-3-5/test_3-5_1-0_326.py::test_add_edge

def test_add_edge():
        g = Graph()
        g.add_node('B')
        g.add_node('C')
>       g.add_edge('B', 'C')

ts-3-5/test_3-5_1-0_326.py:20: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <graph2.Graph object at 0x7f196ab2b400>, n1 = 'B', n2 = 'C'

>   ???
E   NameError: name 'graph' is not defined

graph2:49: NameError
   - [#  49] SVD graph2: 
--------------------------------------------------------------------------------
  49:         self.graph[n1].add(n2)
  50:     
  51:     def del_node(self, n):
  52:         """Delete the node 'n' from the graph."""
- 53:         del self.graph[n]
+ 53:         del graph[n]
  54:         for k in self.graph:
  55:             self.graph[k].discard(n)
  56:     
  57:     def del_edge(self, n1, n2):
--------------------------------------------------------------------------------
[0.21550 s] killed by ts-3-5/test_3-5_1-0_326.py::test_del_node

def test_del_node():
        g = Graph()
        g.add_node('D')
>       g.del_node('D')

ts-3-5/test_3-5_1-0_326.py:26: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <graph2.Graph object at 0x7f196acc4af0>, n = 'D'

>   ???
E   NameError: name 'graph' is not defined

graph2:53: NameError
   - [#  50] SVD graph2: 
--------------------------------------------------------------------------------
  50:     
  51:     def del_node(self, n):
  52:         """Delete the node 'n' from the graph."""
  53:         del self.graph[n]
- 54:         for k in self.graph:
+ 54:         for k in graph:
  55:             self.graph[k].discard(n)
  56:     
  57:     def del_edge(self, n1, n2):
  58:         '''Delete the edge connecting n1 and n2.'''
--------------------------------------------------------------------------------
[0.20689 s] killed by ts-3-5/test_3-5_1-0_326.py::test_del_node

def test_del_node():
        g = Graph()
        g.add_node('D')
>       g.del_node('D')

ts-3-5/test_3-5_1-0_326.py:26: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <graph2.Graph object at 0x7f196ac34d90>, n = 'D'

>   ???
E   NameError: name 'graph' is not defined

graph2:54: NameError
   - [#  51] SVD graph2: 
--------------------------------------------------------------------------------
  51:     def del_node(self, n):
  52:         """Delete the node 'n' from the graph."""
  53:         del self.graph[n]
  54:         for k in self.graph:
- 55:             self.graph[k].discard(n)
+ 55:             graph[k].discard(n)
  56:     
  57:     def del_edge(self, n1, n2):
  58:         '''Delete the edge connecting n1 and n2.'''
  59:         self.graph[n1].remove(n2)
--------------------------------------------------------------------------------
[0.19003 s] survived
   - [#  52] SVD graph2: 
--------------------------------------------------------------------------------
  55:             self.graph[k].discard(n)
  56:     
  57:     def del_edge(self, n1, n2):
  58:         '''Delete the edge connecting n1 and n2.'''
- 59:         self.graph[n1].remove(n2)
+ 59:         graph[n1].remove(n2)
  60:     
  61:     def has_node(self, n):
  62:         """Return boolean if 'n' is in the graph."""
  63:         return n in self.graph
--------------------------------------------------------------------------------
[0.23339 s] killed by ts-3-5/test_3-5_1-0_326.py::test_del_edge

def test_del_edge():
        g = Graph()
        g.add_node('E')
        g.add_node('F')
        g.add_edge('E', 'F')
>       g.del_edge('E', 'F')

ts-3-5/test_3-5_1-0_326.py:34: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <graph2.Graph object at 0x7f196ab2b2b0>, n1 = 'E', n2 = 'F'

>   ???
E   NameError: name 'graph' is not defined

graph2:59: NameError
   - [#  53] SVD graph2: 
--------------------------------------------------------------------------------
  59:         self.graph[n1].remove(n2)
  60:     
  61:     def has_node(self, n):
  62:         """Return boolean if 'n' is in the graph."""
- 63:         return n in self.graph
+ 63:         return n in graph
  64:     
  65:     def neighbors(self, n):
  66:         '''Return the list of all nodes connected to n by edges.'''
  67:         return self.graph[n]
--------------------------------------------------------------------------------
[0.21273 s] killed by ts-3-5/test_3-5_1-0_326.py::test_add_node

def test_add_node():
        g = Graph()
        g.add_node('A')
>       assert g.has_node('A')

ts-3-5/test_3-5_1-0_326.py:14: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <graph2.Graph object at 0x7f196addff70>, n = 'A'

>   ???
E   NameError: name 'graph' is not defined

graph2:63: NameError
   - [#  54] SVD graph2: 
--------------------------------------------------------------------------------
  63:         return n in self.graph
  64:     
  65:     def neighbors(self, n):
  66:         '''Return the list of all nodes connected to n by edges.'''
- 67:         return self.graph[n]
+ 67:         return graph[n]
  68:     
  69:     def adjacent(self, n1, n2):
  70:         '''Return boolean if there is an edge connecting n1 and n2.'''
  71:         return n2 in self.neighbors(n1)
--------------------------------------------------------------------------------
[0.21203 s] killed by ts-3-5/test_3-5_1-0_326.py::test_add_edge

def test_add_edge():
        g = Graph()
        g.add_node('B')
        g.add_node('C')
        g.add_edge('B', 'C')
>       assert g.adjacent('B', 'C')

ts-3-5/test_3-5_1-0_326.py:21: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
graph2:71: in adjacent
    ???
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <graph2.Graph object at 0x7f196ab6cee0>, n = 'B'

>   ???
E   NameError: name 'graph' is not defined

graph2:67: NameError
   - [#  55] SVD graph2: 
--------------------------------------------------------------------------------
  67:         return self.graph[n]
  68:     
  69:     def adjacent(self, n1, n2):
  70:         '''Return boolean if there is an edge connecting n1 and n2.'''
- 71:         return n2 in self.neighbors(n1)
+ 71:         return n2 in neighbors(n1)
--------------------------------------------------------------------------------
[0.21872 s] killed by ts-3-5/test_3-5_1-0_326.py::test_add_edge

def test_add_edge():
        g = Graph()
        g.add_node('B')
        g.add_node('C')
        g.add_edge('B', 'C')
>       assert g.adjacent('B', 'C')

ts-3-5/test_3-5_1-0_326.py:21: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <graph2.Graph object at 0x7f196a8ee6e0>, n1 = 'B', n2 = 'C'

>   ???
E   NameError: name 'neighbors' is not defined

graph2:71: NameError
   - [#  56] ZIL graph2: 
--------------------------------------------------------------------------------
  27:         '''Initialize graph.'''
  28:         self.graph = {}
  29:         if data:
  30:             for i in data:
- 31:                 self.add_node(i)
+ 31:                 break
  32:     
  33:     def nodes(self):
  34:         '''Return a list of all nodes in the graph.'''
  35:         return list(self.graph.keys())
--------------------------------------------------------------------------------
[0.18521 s] survived
   - [#  57] ZIL graph2: 
--------------------------------------------------------------------------------
  51:     def del_node(self, n):
  52:         """Delete the node 'n' from the graph."""
  53:         del self.graph[n]
  54:         for k in self.graph:
- 55:             self.graph[k].discard(n)
+ 55:             break
  56:     
  57:     def del_edge(self, n1, n2):
  58:         '''Delete the edge connecting n1 and n2.'''
  59:         self.graph[n1].remove(n2)
--------------------------------------------------------------------------------
[0.19110 s] survived
[*] Mutation score [13.39342 s]: 53.3%
   - all: 57
   - killed: 24 (42.1%)
   - survived: 21 (36.8%)
   - incompetent: 12 (21.1%)
   - timeout: 0 (0.0%)
